tls/s2n_resume.c

```diff
@@ -495,7 +495,10 @@ S2N_RESULT s2n_store_to_cache(struct s2n_connection *conn)
     RESULT_ENSURE(conn->session_id_len <= S2N_TLS_SESSION_ID_MAX_LEN, S2N_ERR_SESSION_ID_TOO_LONG);
 
     RESULT_GUARD_POSIX(s2n_stuffer_init(&to, &entry));
-    RESULT_GUARD(s2n_resume_encrypt_session_ticket(conn, &to));
+
+    struct s2n_ticket_key *key = s2n_get_ticket_encrypt_decrypt_key(conn->config);
+    RESULT_ENSURE(key != NULL, S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY);
+    RESULT_GUARD(s2n_resume_encrypt_session_ticket(conn, key, &to));
```

Comment:
Nit: Is this ensure necessary? You're not dereferencing `key` here. s2n_resume_encrypt_session_ticket does the derefence and null check.